kv/cursor.go

```diff
@@ -13,9 +13,12 @@ type staticCursor struct {
 }
 
 // Pair is a struct for key value pairs.
+// It also includes an error which should only be non-nil
+// when the Key and Value are nil.
 type Pair struct {
 	Key   []byte
 	Value []byte
+	Err   error
```

Comment:
I missed the use case for an error value on `Pair`. I expected that `ForwardCursor#Err()` would be sufficient to indicate that a read failed and the data should be discarded.

Reply:
I just thought I would leverage Pair type with an Err field as often I am communicating between a goroutine and a cursor somewhat asynchronously.
I could set the err field on the cursor though directly in the goroutine if this is prefered? Just was concerned about a potential race. But we shouldn't be reading via `Err()` before `Next()` returns nil, at which point the goroutine should have exited. Though I guess this could play into the case where a cursor is exited early?
With the `Close()` semantics would you expect it to be called before `Err()` ?